src/main/java/cholog/wiseshop/api/member/service/MemberService.java

```diff
@@ -17,20 +23,40 @@ public class MemberService {
 
     private final MemberRepository memberRepository;
     private final CampaignRepository campaignRepository;
+    private final ProductRepository productRepository;
+    private final OrderRepository orderRepository;
+    private final AddressRepository addressRepository;
 
     public MemberService(
         MemberRepository memberRepository,
-        CampaignRepository campaignRepository
-    ) {
+        CampaignRepository campaignRepository,
+        ProductRepository productRepository, OrderRepository orderRepository,
+        AddressRepository addressRepository) {
         this.memberRepository = memberRepository;
         this.campaignRepository = campaignRepository;
+        this.productRepository = productRepository;
+        this.orderRepository = orderRepository;
+        this.addressRepository = addressRepository;
     }
 
     public void deleteMember(Member member) {
-        List<Campaign> campaigns = campaignRepository.findCampaignByMemberId(member.getId());
+        Long memberId = member.getId();
+        List<Campaign> campaigns = campaignRepository.findCampaignByMemberId(memberId);
         if (campaigns.stream().anyMatch(it -> it.getState().equals(CampaignState.IN_PROGRESS))) {
             throw new WiseShopException(WiseShopErrorCode.MEMBER_INPROGRESS_CAMPAIGN_EXIST);
         }
-        memberRepository.deleteById(member.getId());
+        setEmptyParent(memberId, campaigns);
+        memberRepository.deleteById(memberId);
+    }
+
+    private void setEmptyParent(Long memberId, List<Campaign> campaigns) {
+        List<Product> products = productRepository.findByOwnerId(memberId);
+        List<Order> orders = orderRepository.findByMemberId(memberId);
+        List<Address> addresses = addressRepository.findAllByMemberId(memberId);
+
+        campaigns.forEach(c -> c.setMember(null));
+        products.forEach(p -> p.setOwner(null));
+        orders.forEach(o -> o.setMember(null));
+        addresses.forEach(a -> a.setMember(null));
```

Comment:
모두 복수형이니 단수형으로 하는건 어떨까요??

```suggestion
        campaigns.forEach(campaign -> campaign.setMember(null));
        products.forEach(product -> product.setOwner(null));
        orders.forEach(order -> order.setMember(null));
        addresses.forEach(address -> address.setMember(null));
```